plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-order-downloads-controller.php

```diff
@@ -0,0 +1,267 @@
+<?php
+/**
+ * REST API Order Downloads controller
+ *
+ * Handles requests to the /orders/<order_id>/downloads endpoint.
+ *
+ * @package Automattic/WooCommerce/RestApi
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Order Downloads controller class.
+ *
+ * @package Automattic/WooCommerce/RestApi
+ * @extends WC_REST_Controller
+ */
+class WC_REST_Order_Downloads_Controller extends WC_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'orders/(?P<order_id>[\d]+)/downloads';
+
+	/**
+	 * Post type.
+	 *
+	 * @var string
+	 */
+	protected $post_type = 'shop_order';
+
+	/**
+	 * Register the routes for order downloads.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'order_id' => array(
+						'description' => __( 'The order ID.', 'woocommerce' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Check whether a given request has permission to read order downloads.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! wc_rest_check_post_permissions( $this->post_type, 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
```

Comment:
It may be better to use logic [like this](https://github.com/woocommerce/woocommerce/blob/537cb3c7f7ceecaa968104a0744b96c029d8ee69/plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-crud-controller.php#L52-L60) (since this operation relates to a specific order ID, we can perform a more granular test) and that would avoid the need to [validate the object ID later on (lines 88-92)](https://github.com/woocommerce/woocommerce/pull/27348/files#diff-02a0c38f769112f71851e92f55eed38c3ebc0b527d938ae40a8f35aeda2479ccR88-R92).